### PR TITLE
[Tooling] Add sync operator role script

### LINKED ISSUE #10365: ✨ Sync roles script
## ✨ Feature

Write a repeatable artisan script which goes through our existing Pool Operators and assigns them the Process Operator role on appropriate Pools.

## 🕵️ Details

Add any additional details that could assist with the development of the new feature.

## 🎨 Design File

Include a link to the design file (if it exists).

### 🧑‍🎨 Designer

Include the author of the linked design file's name (or even better, their `@githubUsername`) for future reference.

## 📸 Screenshot

Add a screenshot of the new feature (if possible).

## 🙋‍♀️ Proposed Implementation

For every Pool Operator:
- find the Pool Operator's team
- for Pools belonging to that team:
  - assign the user the Process Operator role on this Pool's team (using the new 1-1 polymorphic relationship, not the legacy Team->multiple pools relationship), if they don't already have the role

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] artisan script which goes through our existing Pool Operators and assigns them the Process Operator role on appropriate Pools
- [ ] script can be run multiple times without issue 
## 🛑 Blockers

Issues which must be completed before this one.

```[tasklist]
### Blocked By
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10349
- [ ] https://github.com/GCTC-NTGC/gc-digital-talent/issues/10302
```

### PR DESCRIPTION
**Pull request by vd1992**
🤖 Resolves #10365

## 👋 Introduction

Adds an artisan command to sync existing pool operator roles with the new process operator role

## 🕵️ Details

I uh, think I got the goal of the issue. Wasn't entirely sure at first

## 🧪 Testing

1. Run command from below multiple times
2. Check teams and role_user tables

## 🚚 Deployment

Run

`php artisan app:sync-pool-process-operator`

### CODE REVIEW

File: api/app/Console/Commands/SyncPoolOperatorWithProcessOperator.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                $teamPoolsId = $teamsPools->pluck('id')->unique()->toArray();
+                array_push($poolIdsToSync, ...$teamPoolsId);
+            }
+
+            $uniquePoolIdsToSync = array_unique($poolIdsToSync);
+
+            // run through each pool id
+            foreach ($uniquePoolIdsToSync as $poolId) {
+                // create or fetch the pool teamable
+                $poolTeam = Team::firstOrCreate(
```
Comment (tristan-orourke) [inline api/app/Console/Commands/SyncPoolOperatorWithProcessOperator.php:64]: I recommend using pool->addProcessOperators here, to help ensure consistency in how teams are created.

Comment (tristan-orourke) [reply api/app/Console/Commands/SyncPoolOperatorWithProcessOperator.php:64]: Though... double check that it doesn't create duplicate roles for the same user-role-team combo, and fix it if it does 😅

**Review by tristan-orourke — CHANGES_REQUESTED** [review]

File: api/app/Console/Commands/SyncPoolOperatorWithProcessOperator.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            // collect array of teams the user is a pool operator on
+            $poolOperatorRoleAssignments = $poolUser->roleAssignments->where('role_id', $poolOperatorRole->id);
+            $poolOperatorTeams = ($poolOperatorRoleAssignments->pluck('team_id')->unique()->toArray());
+
+            // collect the pools per team then build an id array to sync
+            $poolIdsToSync = [];
+            foreach ($poolOperatorTeams as $teamId) {
+                $teamsPools = Pool::whereHas('legacyTeam', function ($query) use ($teamId) {
+                    $query->where('team_id', $teamId);
+                })->get();
```
[the remark this answers is not in the document]
Comment (vd1992) [reply api/app/Console/Commands/SyncPoolOperatorWithProcessOperator.php]: Oh thanks, that seems to work just fine and more condense
https://github.com/GCTC-NTGC/gc-digital-talent/pull/10743/commits/e5e6dacf051fd68c1c8cca4acc339928cdb3d80f

File: api/app/Console/Commands/SyncPoolOperatorWithProcessOperator.php (hunk 1 above)
Comment (vd1992) [reply api/app/Console/Commands/SyncPoolOperatorWithProcessOperator.php]: Ah, that would be a better idea. More condense too.
Seems to work fine with multiple runs, not spotting duplicates in table role_user
https://github.com/GCTC-NTGC/gc-digital-talent/pull/10743/commits/65a7f4992f0b860e3e27a39377ec0f913d89d49c

**Comment by tristan-orourke** [comment]
SQL script for testing:

[select](https://www.postgresql.org/docs/12/static/sql-select.html) [users](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=users).email, [roles](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=roles).name, [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams).name, [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams).teamable_type, [communities](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=communities).key, [pools](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=pools).name->>'en'
FROM [users](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=users)
JOIN [role_user](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=role_user) ON [role_user](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=role_user).user_id = [users](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=users).id
JOIN [roles](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=roles) ON [role_user](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=role_user).role_id = [roles](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=roles).id
LEFT JOIN [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams) ON [role_user](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=role_user).team_id = [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams).id
LEFT JOIN [communities](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=communities) ON [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams).teamable_id = [communities](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=communities).id
LEFT JOIN [pools](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=pools) ON [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams).teamable_id = [pools](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=pools).id[select](https://www.postgresql.org/docs/12/static/sql-select.html) [users](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=users).email, [roles](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=roles).name, [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams).name, [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams).teamable_type, [communities](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=communities).key, [pools](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=pools).name->>'en'
FROM [users](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=users)
JOIN [role_user](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=role_user) ON [role_user](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=role_user).user_id = [users](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=users).id
JOIN [roles](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=roles) ON [role_user](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=role_user).role_id = [roles](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=roles).id
LEFT JOIN [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams) ON [role_user](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=role_user).team_id = [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams).id
LEFT JOIN [communities](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=communities) ON [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams).teamable_id = [communities](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=communities).id
LEFT JOIN [pools](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=pools) ON [teams](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=teams).teamable_id = [pools](http://localhost:8080/?pgsql=postgres&username=postgres&db=gctalent&ns=public&table=pools).id

**Review by tristan-orourke — APPROVED** [review]
Looks great!

**Pull request merged**